app/models/concerns/tube_material.rb

```diff
@@ -3,8 +3,13 @@
 # Material
 module TubeMaterial
```

Comment:
I don't know if you need this any more. Can you instead add something like: `has_one :tube, through: :container_material, source: :container, source_type: 'Tube'` to the material concern that will do the same thing? Or at least add this method to the material concern?

Reply:
As discussed, I think this is a reasonable thing to leave in since it distinguishes materials that expect to only have a tube from those that are aware they have a polymorphic type of container.